optimus_manager/xorg.py

```diff
@@ -79,7 +79,11 @@ def do_xsetup(requested_mode):
         logger.info("Running xrandr commands")
 
         try:
-            exec_bash("xrandr --setprovideroutputsource modesetting NVIDIA-0")
+            provider = checks.get_integrated_provider()
+            if config["amd"]["driver"] == "amdgpu" and "amd" in bus_ids:
+                exec_bash("xrandr --setprovideroutputsource \"%s\" NVIDIA-0" % provider)
+            else:
+                exec_bash("xrandr --setprovideroutputsource modesetting NVIDIA-0")
```

Comment:
If `get_integrated_provider()` always return the exact name of the provider (see above), we can make the if..else block redundant and only keep the "if" branch.